ui/pages/import-token/import-token.test.js

```diff
@@ -1,110 +1,177 @@
 
-      expect(
-        wrapper.find('ImportToken').instance().state.customSymbol,
-      ).toStrictEqual(tokenSymbol);
+      expect(document.getElementById('custom-symbol').value).toStrictEqual(
+        tokenSymbol,
+      );
     });
 
     it('edits token decimal precision', () => {
+      const { getByText } = render();
+      const customTokenButton = getByText('Custom Token');
+      fireEvent.click(customTokenButton);
+
       const tokenPrecision = '2';
       const event = { target: { value: tokenPrecision } };
-      const customAddress = wrapper.find('#custom-decimals');
-      customAddress.last().simulate('change', event);
+      fireEvent.change(document.getElementById('custom-decimals'), event);
 
-      expect(
-        wrapper.find('ImportToken').instance().state.customDecimals,
-      ).toStrictEqual(Number(tokenPrecision));
+      expect(document.getElementById('custom-decimals').value).toStrictEqual(
+        tokenPrecision,
+      );
     });
 
-    it('next', () => {
-      const nextButton = wrapper.find(
-        '.button.btn-primary.page-container__footer-button',
-      );
-      nextButton.simulate('click');
+    it('adds custom tokens successfully', async () => {
+      const { getByText } = render();
+      const customTokenButton = getByText('Custom Token');
+      fireEvent.click(customTokenButton);
 
-      expect(props.setPendingTokens.calledOnce).toStrictEqual(true);
-      expect(props.history.push.calledOnce).toStrictEqual(true);
-      expect(props.history.push.getCall(0).args[0]).toStrictEqual(
-        '/confirm-import-token',
-      );
+      const submit = getByText('Add Custom Token');
+      expect(submit).toBeDisabled();
+
+      const tokenAddress = '0x617b3f8050a0BD94b6b1da02B4384eE5B4DF13F4';
+      fireEvent.change(document.getElementById('custom-address'), {
+        target: { value: tokenAddress },
+      });
+      expect(submit).not.toBeDisabled();
+
+      const tokenSymbol = 'META';
+      fireEvent.change(document.getElementById('custom-symbol'), {
+        target: { value: tokenSymbol },
+      });
+
+      const tokenPrecision = '2';
+      await fireEvent.change(document.getElementById('custom-decimals'), {
+        target: { value: tokenPrecision },
+      });
+
+      expect(submit).not.toBeDisabled();
+      fireEvent.click(submit);
+      expect(setPendingTokens).toHaveBeenCalledWith({
+        customToken: {
+          address: tokenAddress,
+          decimals: Number(tokenPrecision),
+          symbol: tokenSymbol,
+        },
+        selectedTokens: {},
+        tokenAddressList: [],
+      });
+      expect(historyStub).toHaveBeenCalledWith('/confirm-import-token');
     });
 
-    it('cancels', () => {
-      const cancelButton = wrapper.find('.page-container__header-close');
-      cancelButton.simulate('click');
+    it('cancels out of import token flow', () => {
+      const { getAllByRole } = render();
+      const closeButton = getAllByRole('button')[0];
```

Comment:
This seems a little fragile do we not have an aria label on the close button?
`<button aria-label="close" />`
`getByRole('button', { name: 'close' })`

Reply:
I can add one!

Reply:
actually the cancel button [already has a `data-testid`](https://github.com/MetaMask/metamask-extension/blob/cb815a91ad8de86f3aaf0e36959d412b441498e9/ui/components/ui/page-container/page-container-footer/page-container-footer.component.js#L54) so going to use that

Reply:
wait nevermind this test was for the top "x" button. Will add the aria label